src/main/java/io/github/thebusybiscuit/cscorelib2/inventory/InvUtils.java

```diff
@@ -102,7 +123,7 @@ public static boolean fitAll(@NonNull Inventory inv, @NonNull ItemStack[] items,
                 if (stack == null || stack.getType() == Material.AIR) {
                     cache.put(slot, items[i]);
                     resolved = true;
-                } else if (stack.getAmount() + item.getAmount() <= stack.getMaxStackSize() && ItemUtils.canStack(stack, item)) {
+                } else if (underMaxStackSize(stack, item, inv) && ItemUtils.canStack(stack, item)) {
```

Comment:
Need to update that too
```suggestion
            } else if (isValidStackSize(stack, item, inv) && ItemUtils.canStack(stack, item)) {
```